Class1/HouseRobber.java

```diff
@@ -0,0 +1,33 @@
+package Class1;
+
+// https://leetcode.com/problems/house-robber/
+// start: 10:57
+// end and pass all LC test cases: 11:01
+public class HouseRobber {
```

Comment:
> can you solve it again but with recursion + memo?

Reply:
Yes, here you go: https://github.com/zhanwz/CodingExercise/commit/ce9e1f106257a5c981847524b9a13d1a52d38ddc